src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/DelegateCtorSignature.cs

```diff
@@ -15,18 +15,22 @@ public class DelegateCtorSignature : Signature
 
         private readonly IMethodNode _targetMethod;
 
+        private readonly bool _isInstantiatingStub;
+
         private readonly ModuleToken _methodToken;
 
         private readonly SignatureContext _signatureContext;
 
         public DelegateCtorSignature(
             TypeDesc delegateType,
             IMethodNode targetMethod,
+            bool isInstantiatingStub,
```

Comment:
How does this help? It seems like we use `Method.HasInstantiation` at the callsite to populate this, but then we don't really use it for anything (besides when computing the name, which I believe makes the assert you saw go away). `isInstantiatingStub: _targetMethod.Method.HasInstantiation` below is still hardcoded to get the `HasInstantiation` from the method instead of from this new field.

The generic handling here looked pretty suspicious to me so I tried running one of the CoreRT tests on this that I know exercises generics pretty well (it's a test written to exercise AOT compilers, as opposed to the smorgasbord of random tests in the CoreCLR repo that sometimes happen to find AOT bugs). The test fails. I filed #31654.

Reply:
You're right, I have verified this is indeed superfluous, it was a leftover from my earlier experiments at fixing the comparison assertion that later turned out to be caused by the incompleteness in CompareToImpl. I have rolled back this part of my change in 2nd commit.